pyiron_base/job/jobtype.py

```diff
@@ -95,8 +98,48 @@ def __new__(
             job.set_input_to_read_only()
         return job
 
+    @classmethod
+    def unregister(cls, job_name_or_class):
+        if isinstance(job_name_or_class, type):
+            job_name_or_class = job_name_or_class.__name__
+        if job_name_or_class in cls._job_class_dict:
+            del cls._job_class_dict[job_name_or_class]
+        else:
+            raise KeyError(f"No JobType with name '{job_name_or_class}' found.")
+        return cls
```

Comment:
Should return the correct class...
```suggestion
            raise KeyError(f"No JobType with name '{job_name_or_class}' found.")
        return job_name_or_class
```